### DISCUSSION TITLE
How to import custom shapes/components?

### DISCUSSION THREAD

**Discussion by YihaoCui1** [HelgeGehring/gdshelpers#60] [Q&A]
Hi there, is there any way to import custom designed components with this code?

**Comment by HelgeGehring** [comment]
Hey :)
In which format do you have your custom designed components?

You can either generate a shapely polygon and add it to a cell, or write your own part, which returns a shapely-polygon.
Have a look at the scripts for the marker-structures, they are rather short: https://github.com/HelgeGehring/gdshelpers/blob/master/gdshelpers/parts/marker.py

Hope that helps!
Best,
Helge
